### ISSUE TITLE
recipe for differentiable versions K.max() and K.min()

### ISSUE THREAD

**Issue by RichardAFrazin** [keras-team/keras#12854]
feature suggestion:

K.abs() allows you to include a differentiable version of K.max() and K.min().
This is easy because, for scalars a and b, we have:
max(a, b) = .5*( a + b + |a - b |) 

similarly,
min(a, b) = .5*( a + b - |a - b| )

Also, a differentiable switch on the sign of a scalar value b can be implemented as:

(1 + b/|b|) / 2     or (1 - b/|b|) / 2

**Comment by Dref360** [comment]
K.max and K.min are already differentiable, aren't they?

**Comment by RichardAFrazin** [comment]
They are not.

**Comment by Dref360** [comment]
I mean the gradient is not strong and even with your proposition the gradient will be almost 0 everywhere, but if max and min were not differentiable, most activation functions wouldn't work.

**Comment by RichardAFrazin** [comment]
I apologize.  It seems you are correct and you can use K.max() in the loss function, so it must be differentiable.   I was getting an error message that I cannot now reproduce, but the error message listed K.max as a possible cause of having a non-differentiable loss function (even though I was not using K.max).
